vaadin-rich-text-editor-flow-parent/vaadin-rich-text-editor-flow/src/main/java/com/vaadin/flow/component/richtexteditor/RichTextEditor.java

```diff
@@ -152,94 +158,87 @@ public ValueChangeMode getValueChangeMode() {
     @Override
     public void setValueChangeMode(ValueChangeMode valueChangeMode) {
         currentMode = valueChangeMode;
-        setSynchronizedEvent(
-                ValueChangeMode.eventForMode(valueChangeMode, "value-changed"));
+        setSynchronizedEvent(ValueChangeMode.eventForMode(valueChangeMode,
+                "html-value-changed"));
     }
 
     /**
-     * Sets the value of this editor. Should be in
-     * <a href="https://github.com/quilljs/delta">Delta</a> format. If the new
-     * value is not equal to {@code getValue()}, fires a value change event.
-     * Throws {@code NullPointerException}, if the value is null.
+     * Sets the value of this editor in HTML format. If the new value is not
+     * equal to {@code getValue()}, fires a value change event. Throws
+     * {@code NullPointerException}, if the value is null.
      * <p>
      * Note: {@link Binder} will take care of the {@code null} conversion when
      * integrates with the editor, as long as no new converter is defined.
      *
-     * @see #asHtml()
-     * @see AsHtml#setValue(String)
+     * @see #asDelta()
+     * @see AsDelta#setValue(String)
      * @param value
-     *            the new value in Delta format, not {@code null}
-     * @deprecated since 23.3, from 24 onwards the value will be in HTML format.
-     *             Use {@link #asDelta()} instead.
+     *            the new value in HTML format, not {@code null}
      */
     @Override
-    @Deprecated
     public void setValue(String value) {
+        Objects.requireNonNull(value, "Null value is not supported");
         super.setValue(value);
     }
 
-    @ClientCallable
-    private void updateValue(String value) {
-        setValue(value);
-        if (this.asHtml != null) {
-            this.asHtml.value.clear();
-        }
+    @Override
+    protected void setPresentationValue(String newPresentationValue) {
+        getElement().setProperty("htmlValue", newPresentationValue);
+        // htmlValue property is not writeable, HTML value needs to be set using
+        // method exposed by web component instead
+        getElement().callJsFunction("dangerouslySetHtmlValue", this.getValue());
```

Comment:
Looks like because of some timing change related to https://github.com/vaadin/flow-components/pull/4499, it's possible that the call to the function gets executed before the element has been fully upgraded. Some other components that use `callJsFunction` suffer from the same side-effect also.

Reply:
We discussed this internally. The Dialog timing issue fix as suggested by @sissbruecker: https://github.com/vaadin/flow-components/pull/4547